src/snapshotcloneserver/clone/clone_service_manager.cpp

```diff
@@ -231,6 +232,25 @@ int CloneServiceManager::Flatten(
 
     auto cloneInfoMetric = std::make_shared<CloneInfoMetric>(taskId);
     auto closure = std::make_shared<CloneClosure>();
+
+    // init dlock
+    if (!useMockDLock_) {
+        dlockOpts_->pfx = std::to_string(cloneInfo.GetDestId());
+        dlock_ = std::make_shared<DLock>(*dlockOpts_);
+        if (0 == dlock_->Init()) {
+            LOG(ERROR) << "Init DLock error"
+                    << ", pfx = " << dlockOpts_->pfx
+                    << ", retryTimes = " << dlockOpts_->retryTimes
+                    << ", timeout = " << dlockOpts_->timeout
+                    << ", ttl = " << dlockOpts_->ttl;
+            return kErrCodeInternalError;
+        }
+    }
+
+    DLockGuard dlockGuard(dlock_);
+    closure->SetDLock(dlock_);
+    dlockGuard.Release();
```

Comment:
do nothing after lock?

Reply:
> do nothing after lock?

unlock fater task finished in task Run()